pkg/framework/plugins/removepodsviolatingtopologyspreadconstraint/types.go

```diff
@@ -31,4 +31,5 @@ type RemovePodsViolatingTopologySpreadConstraintArgs struct {
 	Namespaces             *api.Namespaces       `json:"namespaces"`
 	LabelSelector          *metav1.LabelSelector `json:"labelSelector"`
 	IncludeSoftConstraints bool                  `json:"includeSoftConstraints"`
+	TopologyBalanceNodeFit *bool                 `json:"TopologyBalanceNodeFit"`
```

Comment:
How about?
```json
RemovePodsViolatingTopologySpreadConstraintArgs:
- TopologyBalanceNodeFit: [] // empty field -> fall back to defaults which is a subject to change
- TopologyBalanceNodeFit: [""] // empty string -> disable the node fit checks
- TopologyBalanceNodeFit: ["Cond1", ..., "CondN"] // arbitrary list of enabled conditions (implemented later)
```

```go
type NodeFitConditions string

var (
	NoCondition NodeFitConditions = ""
)
```

Alternatively, `""` can be replaced with `"NoCheck"` or similar string which if specified will be the only valid item in the list.

Reply:
I was under the impression that we will handle NodeFit Spec in a subsequent design/PR. But given that, perhaps we should design/implement that first.

Reply:
This is more like a middle step so once the NodeFit is refined the API of `TopologyBalanceNodeFit` does not need to change. It will "just" get extended with additional enums.